CMakeLists.txt

```diff
@@ -70,7 +68,7 @@ if( UNIX )
     # WIN32 has '#define interface struct', so we'll do the same thing.
     # This should not affect anything in the code, but it will cause it to not compile on
     # non-Windows systems
-    add_compile_definitions( interface=struct )
+    add_definitions( -Dinterface=struct )
```

Comment:
I don't think this is supported on CMake 3.8. I have CMake 3.10.2 and it was complaining.

Reply:
It looks like `add_compile_definitions` was added in 3.12, so if that's supported on Windows 7 we can probably just set that as the minimum version

Reply:
Sure, going to CMake 3.12 sounds like a solution.

Reply:
bumped CMake to 3.12.